src/test_linked_list.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python
+# -*- coding: utf -8 -*-
+
+from linked_list import Node, LinkedList
+
+
+def test_instance_node():
+    node1 = Node(3)
+    assert node1.value == 3 and node1.next_node is None
+
+
+def test_instance_linked_list():
+    list1 = LinkedList()
+    assert list1.head_node is None
+
+
+def test_linked_list_push():
+    list1 = LinkedList()
+    assert list1.push(3) == list1.head_node
+
+
+def test_linked_list__len__():
+    list1 = LinkedList()
+    list1.push(4)
+    assert list1.__len__() == 1
+
+
+def test_len():
+    list1 = LinkedList()
+    list1.push(4)
+    assert len(list1) == 1
+
+
+def test_linked_list_size_1():
+    list1 = LinkedList()
+    list1.push(4)
+    list1.push(2)
+    list1.pop()
+    assert list1.size() == 1
```

Comment:
After the value is popped out, you should test that the value no longer exists in the list.